tests/integration/devfile/cmd_dev_test.go

```diff
@@ -164,6 +166,51 @@ var _ = Describe("odo dev command tests", func() {
 			}, 180, 10).Should(Equal(true))
 
 		})
+	})
 
+	Context("port-forwarding for the component", func() {
```

Comment:
A few scenarios that could be added:
* One component with one endpoint is pushed and its endpoint is forwarded to port 40001. When another component is pushed after that, odo sets up port-fowarding on port 40002 for its endpoint. In a way, this is already being tested by the scenario for devfile with multiple endpoints
* A devfile with no endpoints set to `exposure: public` or `exposure: internal`. In a way, this is also tested by multiple endpoint spec because the devfile there has three endpoints with each exposure value - public, internal and none.